test/sql/secrets/create_secret_storage_backends.test

```diff
@@ -21,7 +21,7 @@ Can not set secret storage for temporary secrets!
 statement error
 CREATE PERSISTENT SECRET s1 IN NON_EXISTENT_SECRET_STORAGE ( TYPE S3 )
 ----
-Secret storage 'non_existent_secret_storage' not found!
```

Comment:
nit: can we maybe test both cases? 

So running `CREATE PERSISTENT SECRET s1 IN NON_EXISTENT_SECRET_STORAGE ( TYPE S3 )` with `allow_persistent_secrets=false;` should result in `Persistent secrets are currently disabled. To enable them, restart duckdb and run 'SET allow_persistent_secrets=true'` 

whereas running it with `allow_persistent_secrets=true` should print `Secret storage 'non_existent_secret_storage' not found!`

Reply:
Actually, the test failure in this [CI build](https://github.com/duckdb/duckdb/actions/runs/7480844424/job/20363757014) prompted me to examine the logic in a bit more details and I pushed another change. Essentially, I think we need to consider both `persistence_type` coming from the SQL statement and also the resolved storage backend's own `persistent` parameter. In addition, I think loading a persistent secret backend is orthogonal to `allow_persistent_secrets` since the latter is only relevant when you create a secret. I updated test cases accordingly too.